### DISCUSSION TITLE
Writing a formula for my own repository

### DISCUSSION THREAD

**Discussion by rcdailey** [Homebrew/discussions#4201] [Writing Formulae/Casks] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 3.6.21
ORIGIN: https://github.com/Homebrew/brew
HEAD: 932d2cf3b77c9439a57b6a43577fc8d3b6399a62
Last commit: 5 days ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: deecd6afc1ed2ce43805fc31cc1d0d2724ba624d
Core tap last commit: 6 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: dodeca-core 64-bit arm_blizzard_avalanche
Clang: 14.0.0 build 1400
Git: 2.39.1 => /opt/homebrew/bin/git
Curl: 7.86.0 => /usr/bin/curl
macOS: 13.2-arm64
CLT: 14.2.0.0.1.1668646533
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

I'm the developer of [Recyclarr](https://github.com/recyclarr/recyclarr) and I wanted to create a brew formula for my application, which already has precompiled binaries attached to each Github Release.

I read the "Acceptable Formulae" page, and it says [no self-submitted](https://docs.brew.sh/Acceptable-Formulae#niche-or-self-submitted-stuff) packages unless they are "very popular". My question is: What qualifies as "very popular"?

I should probably take a step back and ask *where* a self-submitted formula, like mine, would go. I assumed the official tap(s) like the core one would be where it goes. If my assumption is wrong, then I guess my previous question is invalidated.

Could someone let me know if the core tap is where my application should go? And if not, which tap should I use? I don't think I want to host my own because that seems like it would require users to add a custom tap to search for my application. I did also read in the docs that the casks tap is where it should go, but my understanding is that casks are reserved for applications with a GUI. Mine is a CLI app.

I'm still learning so I apologize if my understanding is completely off!

**Comment by p-linnane** [comment, marked as answer]
A CLI app that can be built from source would go in `homebrew-core`. If you only distribute precompiled binaries, that would go into `homebrew-cask`.

**Reply by rcdailey** [reply]
Whether it's a GUI or CLI app doesn't matter, it would still be a cask? Thanks for your help.

**Reply by p-linnane** [reply]
Yes. All applications in `homebrew-core` must build from source, and that source must be licensed under a license listed in the [Debian Free Software Guidelines](https://wiki.debian.org/DFSGLicenses).
